rts/Lua/LuaOpenGL.cpp

```diff
@@ -1040,6 +1062,41 @@ inline void LuaOpenGL::CheckDrawingEnabled(lua_State* L, const char* caller)
 	}
 }
 
+inline void LuaOpenGL::CondWarnDeprecatedGL(lua_State* L, const char* caller)
+{
+	if (deprecatedGLWarnLevel <= 0)
+		return;
+
+	std::string luaCaller;
+
+	if (deprecatedGLWarnLevel >= 2) {
+		lua_Debug info;
+		const int level = 1; // A calling function from Lua
+		if (lua_getstack(L, level, &info)) {
+			lua_getinfo(L, "nSl", &info);
+
+			luaCaller = fmt::format("[{}]:{} Caller: {}", info.short_src, info.currentline, (info.name ? info.name : "<unknown>"));
+		}
+	}
+
+	const auto key = fmt::format("{}{}", caller, luaCaller.c_str());
+
+	if (deprecatedGLWarned.find(key) == deprecatedGLWarned.end()) {
```

Comment:
early exit

Reply:
Do you mean: "reverse the condition and quit if true" ?

Reply:
yes